### PR TITLE
Add method asBool to Clock.

### PR DESCRIPTION
**Pull request by chick**
**Related issue**: 
Addresses Chisel issue #1238

**Type of change**: feature request

**Impact**: Minor API addition (no impact on existing code)

**Development Phase**:  implementation

**Release Notes**
There is now a short cut method for turning a clock to a bool
`clock.asBool`. Previously had to do `clock.asUInt.asBool`

### CODE REVIEW

File: chiselFrontend/src/main/scala/chisel3/Clock.scala [hunk 1]
```diff
@@ -30,9 +31,13 @@ sealed class Clock(private[chisel3] val width: Width = Width(1)) extends Element
...
 
+  /** Returns the contents of this wire as a [[scala.collection.Seq]] of [[Bool]]. */
+  final def asBool(): Bool = macro SourceInfoTransform.noArg
+  def do_asBool(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): Bool = this.asUInt().asBool()
+
   override def do_asUInt(implicit sourceInfo: SourceInfo, connectCompileOptions: CompileOptions): UInt = pushOp(DefPrim(sourceInfo, UInt(this.width), AsUIntOp, ref)) // scalastyle:ignore line.size.limit
   private[chisel3] override def connectFromBits(that: Bits)(implicit sourceInfo: SourceInfo,
       compileOptions: CompileOptions): Unit = {
-    this := that.asBool.asClock
+    this := that.asBool().asClock
```
Comment (jackkoenig) [inline chiselFrontend/src/main/scala/chisel3/Clock.scala]: Why are you adding parentheses here?

Comment (chick) [reply chiselFrontend/src/main/scala/chisel3/Clock.scala]: Accidental, removed them

File: src/test/scala/chiselTests/Clock.scala [hunk 2]
```diff
@@ -18,6 +19,8 @@ class WithClockAndNoReset extends RawModule {
   val a = withClock(clock2) {
     RegNext(in)
   }
+
+  in.asUInt()
```
Comment (ucbjrl) [inline src/test/scala/chiselTests/Clock.scala]: What is the purpose of this cast?

Comment (chick) [reply src/test/scala/chiselTests/Clock.scala]: None, remove it. Thanks for catching

File: chiselFrontend/src/main/scala/chisel3/Clock.scala [hunk 3]
```diff
@@ -30,6 +31,10 @@ sealed class Clock(private[chisel3] val width: Width = Width(1)) extends Element
   /** Not really supported */
   def toPrintable: Printable = PString("CLOCK")
 
+  /** Returns the contents of this wire as a [[scala.collection.Seq]] of [[Bool]]. */
```
Comment (ucbjrl) [inline chiselFrontend/src/main/scala/chisel3/Clock.scala]: I'm confused. `asBool` returns a `Seq`? Shouldn't the method be `asBools`?

Comment (aswaterman) [reply chiselFrontend/src/main/scala/chisel3/Clock.scala]: I think the method is as intended, but the comment is wrong

Comment (ucbjrl) [reply chiselFrontend/src/main/scala/chisel3/Clock.scala]: I've updated the comment. If others approve, so do I.

**Pull request merged**